### PR TITLE
MSC2174: Move the `redacts` key to a sane place

### LINKED ISSUE #502: `m.room.redaction` has `redacts` at the event level (SPEC-164)
Submitted by @&#8203;kegan:matrix.org
`m.room.redaction` has the redacts key at the top level of the event. Is this really necessary? Given nearly every other event doesn't do this, it would be great to move it to content.redacts instead.

(Imported from https://matrix.org/jira/browse/SPEC-164)

### PR DESCRIPTION
**Pull request by richvdh**
[Rendered](https://github.com/matrix-org/matrix-doc/blob/rav/proposals/move_redacts_key/proposals/2174-move-redacts-key.md).

This would fix #502.

### CODE REVIEW

**Comment by KitsuneRal** [comment]
I suppose this change can be bundled with other stuff that would need a new room version. From the client perspective I don't see a big problem (though that will probably be the first case when I'll have to parse events differently depending on which room, i.e. of which version, they arrive to).

File: proposals/2174-move-redacts-key.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# MSC2174: move the `redacts` property to `content`
```
Comment (richvdh) [inline proposals/2174-move-redacts-key.md]: (switching to a comment on the diff, to enable threading)

@KitsuneRal wrote: 

> though that will probably be the first case when I'll have to parse events differently depending on which room, i.e. of which version, they arrive to

Yeah. A workaround for that would be for servers to inject a `redacts` key at the event level when serving `m.room.redaction` events over the C-S api. It's a bit ugly though.

Comment (KitsuneRal) [reply proposals/2174-move-redacts-key.md]: Given that redaction should probably be more accessible than other features, I wouldn't mind a bit of ugliness. The question is whether we should push people into keeping client-side world up-to-date (up-to-half-a-year at least, I suppose)?

Comment (richvdh) [reply proposals/2174-move-redacts-key.md]: I've added some compatibility hacks. I hope they don't prove contentious.

**Comment by richvdh** [comment]
I'd quite like to call FCP on this soonish. It seems relatively uncontentious.

**Comment by richvdh** [comment]
@mscbot fcp merge

**Comment by jplatte** [comment]
So with this, all the necessary data for a redaction event can be sent using the regular message sending endpoint rather than the redaction endpoint. Is that something homeservers are supposed to handle (with the next room version)?

**Comment by turt2live** [comment]
@jplatte this hasn't been assigned to a room version yet, and at a current rate not in time for v6. Best to track the room version proposals (like [v6's](https://github.com/matrix-org/matrix-doc/pull/2240)) for that information.

As I understand it, yes: servers should be able to support boring `/send` calls for redaction events.

**Comment by turt2live** [comment]
Partial implementation (hopefully enough to prove it's feasible?): https://github.com/matrix-org/synapse/pull/12634

**Comment by turt2live** [comment]
Spec PR: https://github.com/matrix-org/matrix-spec/pull/1604

**Comment by turt2live** [comment]
Merged 🎉

**Pull request merged**